images/singleuser-sample/Dockerfile

```diff
@@ -1,26 +1,61 @@
-FROM jupyter/base-notebook:latest
-# Built from... https://hub.docker.com/r/jupyter/base-notebook/
-#               https://github.com/jupyter/docker-stacks/blob/HEAD/base-notebook/Dockerfile
-# Built from... Ubuntu 20.04
+# The build stage
+# ---------------
+FROM python:3.9-bullseye as build-stage
 
-# VULN_SCAN_TIME=2022-08-08_07:54:05
+# VULN_SCAN_TIME=
+
+WORKDIR /build-stage
+
+# Build wheels for packages that require gcc or other build dependencies and
+# lack wheels either for amd64 or aarch64.
+COPY requirements.txt requirements.txt
+RUN pip wheel -r requirements.txt
+
+# The final stage
+# ---------------
+FROM python:3.9-slim-bullseye
+
+# VULN_SCAN_TIME=
+
+ENV DEBIAN_FRONTEND=noninteractive \
+    LANG=en_US.UTF-8 \
+    LC_ALL=en_US.UTF-8 \
+    LANGUAGE=en_US.UTF-8 \
+    NB_USER=jovyan \
+    NB_UID=1000 \
+    HOME=/home/jovyan
 
 USER root
 RUN apt-get update \
  && apt-get upgrade -y \
  && apt-get install -y --no-install-recommends \
+        ca-certificates \
         dnsutils \
         git \
+        locales \
         iputils-ping \
- && rm -rf /var/lib/apt/lists/*
-USER $NB_USER
+        tini \
+ && apt-get clean \
```

Comment:
```suggestion
```

`apt-get clean` is redundant in this situation I've learned from reading Dockerfile best-practices etc. I want to remove it to avoid introducing a hint that it is needed in other Dockerfiles etc even though its probably also harmless to have.